### PR TITLE
Made Item->setLore() fluent

### PR DESCRIPTION
**Pull request by jasonw4331**
## Introduction

setLore() does not return the modified Item instance
### API changes

Item::setLore() now returns Item
### Behavioural changes

No changes
## Backwards compatibility

Full compatability
## Follow-up

None
## Tests

Untested, but should not effect anything due to the returned value not being used by the core.

### CODE REVIEW

**Review by Sandertv — APPROVED** [review]
Good idea, matches with the other set functions from items. This should make life a lot easier.

**Comment by dktapps** [comment]
For reference, this is called fluency, not concurrency.

**Review by matcracker — CHANGES_REQUESTED** [review]
Read my comment

**Comment by TheDiamondYT1** [comment]
@matcracker 
```php
$item = Item::get(Item::GRASS);
$item->setLore(["Lore"])->setCustomName("Lol");
```
```

**Comment by matcracker** [comment]
@TheDiamondYT1 Obviously the gives null...
You should do in this way (remember that is OOP) :
```
$item = Item::get(Item::GRASS);
$item->setLore(["Description"]);
$item->setCustomName("CustomName");
$item->setCount(64);
```

**Comment by TheDiamondYT1** [comment]
@matcracker Of course you can. You can also do it this way:
```php
// mlg performance
die();
```
As @dktapps mentioned, [it's called fluency](https://en.m.wikipedia.org/wiki/Fluent_interface).

**Comment by jasonwynn10** [comment]
If all is good, can this be merged? @dktapps

**Comment by jasonwynn10** [comment]
@matcracker 
I found the non-fluent method to be annoying due to this:
```
$inv = [
Item::get(Item::ARROW)->setCustomName("Name"),
Item::get(Item::WOOL,7,1)->setLore(["Return"])
]
```
this outputs an error

File: src/pocketmine/item/Item.php [hunk 1]
```diff
@@ -687,6 +687,11 @@ public function getLore() : array{
 		return [];
 	}
 
+	/**
+	 * @param array $lines
```
Comment (dktapps) [inline src/pocketmine/item/Item.php]: this should more accurately be `string[]`, not `array`.

**Comment by dktapps** [comment]
Your CI build has failed due to the fact you've included an outdated submodule in your changes. Please rebase your branch onto master and fix your submodule versions.

Protip: don't use `git add *` without being 100% certain what's going to get staged.

**Comment by dktapps** [comment]
You have still included a modified submodule version, please remove the change.

**Comment by jasonwynn10** [comment]
git has been fixed! 😄
Finally ready to merge? 😆

**Review by dktapps — APPROVED** [review]

**Pull request merged**